packages/minter-contracts/test-hs/Test/Swaps/Burn.hs

```diff
@@ -0,0 +1,407 @@
+      let alice ::< bob ::< SNil = sAddresses setup
+      let tokenId1 ::< tokenId2 ::< tokenId3 ::< SNil = sTokens setup
+      (swap, admin) <- originateAllowlistedBurnSwapWithAdmin
+      fa2_1 <- originateFA2 "fa2-1" setup [swap]
+      fa2_2 <- originateFA2 "fa2-2" setup [swap]
+
+      withSender admin $
+        call swap (Call @"Update_allowed") (mkAllowlistSimpleParam [fa2_1, fa2_2])
+
+      assertingBalanceDeltas fa2_1
+        [ (alice, tokenId1) -: -100
+        , (alice, tokenId2) -: -50
+        , (nullAddress , tokenId3) -: 1
+        , (bob, tokenId1) -: 100
+        , (bob, tokenId2) -: 50
+        , (bob, tokenId3) -: -1
+        ] $
+        assertingBalanceDeltas fa2_2
+        [ (alice, tokenId1) -: -1000
+        , (bob, tokenId1) -: 1000
+        ] $ do
+          withSender alice $
+            call swap (Call @"Start") $ mkSingleOffer SwapOffer
+              { assetsOffered =
+                  [ mkFA2Assets fa2_1
+                    [ (tokenId1, 100)
+                    , (tokenId2, 50)
+                    ]
+                  , mkFA2Assets fa2_2
+                    [ (tokenId1, 1000)
+                    ]
+                  ]
+              , assetsRequested =
+                  [ mkFA2Assets fa2_1
+                    [ (tokenId3, 1)
+                    ]
+                  ]
+              }
+          withSender bob $
+            call swap (Call @"Accept") (SwapId 0)
+
+  ]
+
+test_AdminChecks :: TestTree
+test_AdminChecks =
+  adminOwnershipTransferChecks originateAllowlistedBurnSwap
+
+test_AllowlistUpdateAuthorization :: [TestTree]
+test_AllowlistUpdateAuthorization =
+  allowlistUpdateAuthorizationChecks originateAllowlistedBurnSwap
+
+test_AllowlistChecks :: [TestTree]
+test_AllowlistChecks = allowlistSimpleChecks
+  AllowlistChecksSetup
+  { allowlistCheckSetup = \fa2Setup -> do
+      let alice ::< _ = sAddresses fa2Setup
+      let tokenId ::< _ = sTokens fa2Setup
+      (swap, admin) <- originateAllowlistedBurnSwapWithAdmin
+      return (admin, swap, (alice, tokenId))
+
+  , allowlistRestrictionsCases = fromList
+      [ AllowlistRestrictionCase
+        { allowlistError = errSwapOfferedNotAllowlisted
+        , allowlistRunRestrictedAction = \(alice, tokenId) swap (fa2, _) ->
+            withSender alice $
+              call swap (Call @"Start") $ mkSingleOffer SwapOffer
+                { assetsOffered = [mkFA2Assets fa2 [(tokenId, 1)]]
+                , assetsRequested = []
+                }
+        }
+      , AllowlistRestrictionCase
+        { allowlistError = errSwapRequestedNotAllowlisted
+        , allowlistRunRestrictedAction = \(alice, tokenId) swap (fa2, _) ->
+            withSender alice $
+              call swap (Call @"Start") $ mkSingleOffer SwapOffer
+                { assetsOffered = []
+                , assetsRequested = [mkFA2Assets fa2 [(tokenId, 1)]]
+                }
+        }
+      ]
+
+  , allowlistAlwaysIncluded = \_ -> []
+  }
+
+assertingBurnAddressUnchanged
```

Comment:
where is this called?

Reply:
Good catch, I added the assertion into the property tests